pkg/cmd/server/start/start_api.go

```diff
@@ -0,0 +1,76 @@
+package start
+
+import (
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/golang/glog"
+	"github.com/spf13/cobra"
+
+	kerrors "k8s.io/kubernetes/pkg/api/errors"
+	kcmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+)
+
+const apiLong = `Start the master controllers
+
+This command starts the controllers for the master.  Running
+
+  $ %[1]s start master controllers
+
+will start the controllers that manage the master state, including the scheduler. The controllers
+will run in the foreground until you terminate the process.`
+
+// NewCommandStartMasterAPI starts only the APIserver
+func NewCommandStartMasterAPI(fullName string, out io.Writer) (*cobra.Command, *MasterOptions) {
+	options := &MasterOptions{Output: out}
+	options.DefaultsFromName(fullName)
+
+	cmd := &cobra.Command{
+		Use:   "api",
+		Short: "Launch master API",
+		Long:  fmt.Sprintf(apiLong, fullName),
+		Run: func(c *cobra.Command, args []string) {
+			if err := options.Complete(); err != nil {
+				fmt.Fprintln(c.Out(), kcmdutil.UsageError(c, err.Error()))
+				return
+			}
+
+			if len(options.ConfigFile) == 0 {
+				fmt.Fprintln(c.Out(), kcmdutil.UsageError(c, "--config is required for this command"))
+				return
+			}
+
+			if err := options.Validate(args); err != nil {
+				fmt.Fprintln(c.Out(), kcmdutil.UsageError(c, err.Error()))
+				return
+			}
+
+			startProfiler()
+
+			if err := options.StartMaster(); err != nil {
+				if kerrors.IsInvalid(err) {
+					if details := err.(*kerrors.StatusError).ErrStatus.Details; details != nil {
+						fmt.Fprintf(c.Out(), "Invalid %s %s\n", details.Kind, details.Name)
+						for _, cause := range details.Causes {
+							fmt.Fprintf(c.Out(), "  %s: %s\n", cause.Field, cause.Message)
+						}
+						os.Exit(255)
+					}
+				}
+				glog.Fatal(err)
+			}
+		},
+	}
+
+	options.MasterArgs = NewDefaultMasterArgs()
+	options.MasterArgs.StartControllers = false
+
+	flags := cmd.Flags()
+	flags.StringVar(&options.ConfigFile, "config", "", "Location of the master configuration file to run from. Required")
```

Comment:
Down the road, maybe.  For now, I'd like to keep it simple.

On Thu, Aug 13, 2015 at 10:47 AM, David Eads <notifications@github.com>
wrote:

> In pkg/cmd/server/start/start_api.go
> <https://github.com/openshift/origin/pull/4132#discussion_r36982584>:
>
> > +						for _, cause := range details.Causes {
> > +							fmt.Fprintf(c.Out(), "  %s: %s\n", cause.Field, cause.Message)
> > +						}
> > +						os.Exit(255)
> > +					}
> > +				}
> > +				glog.Fatal(err)
> > +			}
> > +		},
> > +	}
> > +
> > +	options.MasterArgs = NewDefaultMasterArgs()
> > +	options.MasterArgs.StartControllers = false
> > +
> > +	flags := cmd.Flags()
> > +	flags.StringVar(&options.ConfigFile, "config", "", "Location of the master configuration file to run from. Required")
>
> Do we want to require a full MasterConfig? Not all of the information is
> required just to start the API server.
>
> Also, there are likely to be commands related to masters like,
> "validate-config", and "write-config". Would we prefer openshift master
> start|[config write|validate|migrate|view-with-defaulting]?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/4132/files#r36982584>.
>



-- 
Clayton Coleman | Lead Engineer, OpenShift